### PR TITLE
[Cpp worker][Java worker]Support Java call Cpp Actor

### PR DESCRIPTION
**Pull request by WangTaoTheTonic**
## Why are these changes needed?
Support java call cpp worker.

todo: doc (will write them all with x-lang invocation)

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by WangTaoTheTonic** [comment]
Failed tests are unrelated.

**Review by raulchen — APPROVED** [review]

File: java/runtime/src/main/java/io/ray/runtime/functionmanager/CppFunctionDescriptor.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import java.util.List;
+
+/** Represents metadata of a Cpp function. */
+public class CppFunctionDescriptor implements FunctionDescriptor {
+
+  public String createFunctionName;
+
+  public String className;
+
+  public String caller;
```
[the remark this answers is not in the document]
Comment (WangTaoTheTonic) [reply java/runtime/src/main/java/io/ray/runtime/functionmanager/CppFunctionDescriptor.java:15]: It's used here: https://github.com/ray-project/ray/blob/master/cpp/src/ray/runtime/task/task_executor.cc#L146

**Comment by WangTaoTheTonic** [comment]
@edoakes @ericl @richardliaw 
Hello, this PR adds support for invoking Cpp actor from Java side, and it adds a file `CppActorCall.java`, generated by `RayCallGenerator.java`, so it need to be excluded from java checkstyle in `ci/lint/format.sh`.
It needs approve from one of you three owners.
Please help to check, thanks !

**Review by richardliaw — APPROVED** [review]
approve for excludes

**Review by edoakes — APPROVED** [review]
Stamp for lint changes.

**Pull request merged**